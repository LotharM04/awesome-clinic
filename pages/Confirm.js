import React, { useEffect, useContext } from "react";
import { AppContext } from "../context/context";
import { StyleSheet, Text, View, Image, TouchableOpacity } from "react-native";
import { DURATION, COLORS, height } from "../assets/styles/theme";
import { appointments, profile } from "../configs/appData";
import Button from "../components/Button";
import * as Animatable from "react-native-animatable";

const ITEM_HEIGHT = height * 0.18;

export default function Confirm({ navigation }) {
  return (
    <View
      style={[
        styles.container,
        {
          flexDirection: "column",
        },
      ]}
    >
      <View style={styles.image}>
        <Animatable.View animation="fadeIn" delay={DURATION + 1 * 100}>
          <Image
            source={require("../assets/icons/check.png")}
            style={styles.imageSVG}
          />
        </Animatable.View>
      </View>
      <View style={{ flex: 2 }}>
        <Animatable.View animation="fadeIn" delay={DURATION + 2 * 100}>
          <Text
            style={{
              color: COLORS.primary,
              textAlign: "center",
              fontSize: 30,
              padding: 10,
              fontWeight: "500",
            }}
          >
            DELETED
          </Text>
          <Text
            style={{
              color: "black",
              textAlign: "center",
              fontSize: 25,
              padding: 10,
            }}
          >
            You have successfully deleted your appointment.
          </Text>
        </Animatable.View>
      </View>
      <View style={{ flex: 1 }}>
        <Animatable.View
          animation="fadeInLeft"
          delay={DURATION + 3 * 100}
          style={styles.btn}
        >
          <TouchableOpacity
            onPress={() => {
              navigation.navigate("Appointments", { appointments, profile });
            }}
          >
            <Button color={COLORS.primary} title={"BACK TO APPOINTMENTS"} />
          </TouchableOpacity>
        </Animatable.View>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
  },
  image: {
    flex: 3,
    alignItems: "center",
    justifyContent: "center",
  },
  imageSVG: {
    width: ITEM_HEIGHT * 1.3,
    height: ITEM_HEIGHT * 1.3,
  },
  btn: {
    alignItems: "center",
  },
});
