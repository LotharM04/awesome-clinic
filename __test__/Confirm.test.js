import React from "react";
import renderer from "react-test-renderer";
import Confirm from "../pages/Confirm";

test("Confirm page renders correctly", () => {
  const tree = renderer.create(<Confirm />).toJSON();
  expect(tree).toMatchSnapshot();
});
