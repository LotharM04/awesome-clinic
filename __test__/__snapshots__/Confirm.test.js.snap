// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm page renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "padding": 20,
      },
      Object {
        "flexDirection": "column",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
        "justifyContent": "center",
      }
    }
  >
    <View
      forwardedRef={[Function]}
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/icons/check.png",
          }
        }
        style={
          Object {
            "height": 312.156,
            "width": 312.156,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <View
      forwardedRef={[Function]}
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#00BFA6",
            "fontSize": 30,
            "fontWeight": "500",
            "padding": 10,
            "textAlign": "center",
          }
        }
      >
        DELETED
      </Text>
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 25,
            "padding": 10,
            "textAlign": "center",
          }
        }
      >
        You have successfully deleted your appointment.
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      forwardedRef={[Function]}
      style={
        Object {
          "alignItems": "center",
          "opacity": 0,
          "transform": Array [
            Object {
              "translateX": -100,
            },
          ],
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          data-testid="button"
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 20,
                "elevation": 3,
                "justifyContent": "center",
                "marginTop": 20,
                "paddingHorizontal": 30,
                "paddingVertical": 10,
              },
              Object {
                "backgroundColor": "#00BFA6",
                "width": undefined,
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
                "letterSpacing": 0.25,
                "lineHeight": 21,
              }
            }
          >
            BACK TO APPOINTMENTS
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
